### PR TITLE
Began splitting given HTML files into various React.JS files

### PR DESCRIPTION
**Pull request by Gabetn**
finished integration for what exists in the case of Login.js, will do Manager.js as well and then leave the rest of the files to the UI team.

### CODE REVIEW

**Review by Gabetn — COMMENTED** [review]
### TODO
#### Splitting of common functionality across different views
 I.e. how should we handle the common components, E.g.  Viewing menu/tags are both visible to manager and customer view. Rather than duplicating the code in both (_Manager.js & Customer.js_), cross view functionality should be consolidated in one place and referenced from the corresponding view.js files. 
 I will look into the viability of using App.js for this purpose. 
**_EDIT_:** Some classes are generic enough to be shared, whilst making reference to specific implementations of features (_See MenuItemList vs MenuItem below_). App.js seems like a good place to hold our exported components (e.g. App)

#### App.js
* Move _CreateDialog_ to **_Manager.js_**
* Share/Export Classes: (_MenuItemList, MenuItem, TagList, Tag_)
**_EDIT_:** _MenuItemList_, just renders a list of _MenuItem_'s, this can be shared between manager and customer, however both will need to have their own overridden versions of _MenuItem_ as the manager is allowed to edit them, whereas the customer can only view. This will of course depend on the UI implementation. 
**Note:** The same can be said for _TagList_ & _Tag_

**Comment by EBruchet** [comment]
- Likely won't move CreateDialog _**Manager.js**_ in a 1:1 fashion, but will create the corresponding view for it - hopefully the functionality as well by today.
- Could potentially create a `Shared.js` file where all these shared components are, but in the case of a MenuItemList, while all it does is render out MenuItems, the styling of it is different dependent on the page. So it would need a page-specific class, which it couldn't have if they all share the same component. Open to suggestion though.

**Comment by Gabetn** [comment]
Yeah, realized last night that **shared.js** won't work due to both view and functionality differing across user type. 
As for _createDialog_ it has to be refactored to match the view for managers since popups aren't used. But since it is only pertinent to Managers I thought it could be moved there, but it could stay on App.js, but that might make it confusing to figure out where code is located

**Comment by EBruchet** [comment]
Nah I was agreeing with your previous remark about moving it, but it won't be the exact same component (not 1:1 transition) as it's not a popup, as you're saying.

**Comment by Gabetn** [comment]
## Bugs
### Styling
I found a issue with the display styling for the _**UpdateDialog**_.  See:
image

**Note:** This was found on _Google Chrome_, on _Firefox_ & _Edge_ the issue was not present (styling was ok), it might just be my specific chrome browser (doubt it though), so try to replicate it on yours. 

### Manual Refresh Requirement
When creating a tag, a manual refresh is required to update the list of tags in the **Tag Search Bar** in _manager_. 
I  will look into this tonight, might be a simple fix.

## Future proofing
### Scalability
Are we going to worry about scaling of our displays?
This is low priority, and should be modified at the end. 
 When attempting to run the app:
**Mobile:**
image

**Tablet**
image

File: src/main/js/components/subcomponents/MenuItem.js [hunk 1]
```diff
@@ -76,37 +104,76 @@ class ManagerMenuItemList extends React.Component {
...
+                response => {
+                    if (response.status !== 200) {
+                        console.log('Looks like there was a problem. Status Code: ' +
+                            response.status);
+                        return;
+                    }
+
+                    // Examine the text in the response
+                    response.json().then((data) => {
+                        console.log(data._embedded.tags);
```
Comment (Gabetn) [inline src/main/js/components/subcomponents/MenuItem.js:181]: Was this meant to be temporary for debugging?

Comment (EBruchet) [reply src/main/js/components/subcomponents/MenuItem.js:181]: Yeah it was, and also on the mobile thing, as you suggest I don't think it's worth looking into - too low priority.

**Pull request merged**